worldedit-core/src/main/java/com/sk89q/worldedit/UnknownDirectionException.java

```diff
@@ -19,20 +19,23 @@
 
 package com.sk89q.worldedit;
 
+import com.sk89q.worldedit.util.formatting.text.TextComponent;
+import com.sk89q.worldedit.util.formatting.text.TranslatableComponent;
+
 /**
  * Thrown when an unknown direction is specified or detected.
  */
 public class UnknownDirectionException extends WorldEditException {
 
-    private String dir;
+    private final String dir;
 
     /**
      * Create a new instance.
      *
      * @param dir the input that was tried
      */
     public UnknownDirectionException(String dir) {
-        super("Unknown direction: " + dir);
+        super(TranslatableComponent.of("worldedit.error.unknown-direction", TextComponent.of(dir)));
```

Comment:
Do we want to make direction translatable too?

Reply:
This is thrown both for user input and for a matched direction which is invalid (eg a diagonal where they aren't accepted).
Perhaps that should actually be split into two different things with different messages?
The issue I see with translating the direction enum values is that the converters won't actually accept localized names as input, making it a bit misleading.

Reply:
Yeah I don't think making it localised makes sense when it's basically a constant seen by the parser